addons/sys_intercom/fnc_infantryPhoneChildrenActions.sqf

```diff
@@ -30,129 +33,107 @@ if (_target isKindOf "CAManBase") then {
+                        [_target, _player, 1, _intercomNetwork] call FUNC(updateInfantryPhoneStatus)
+                    },
+                    {
+                        params ["_target", "_player", "_params"];
+                        _params params ["_intercomNetwork"];
+                        private _isCalling = _target getVariable [QGVAR(isInfantryPhoneCalling), [false, INTERCOM_DISCONNECTED]];
+                        !(_isCalling select 0) || ((_isCalling select 0) && ((_isCalling select 1) == _intercomNetwork))
+                    },
+                    {},
+                    _forEachIndex
+                ] call ace_interact_menu_fnc_createAction;
+                _actions pushBack [_action, [], _target];
+            } forEach _intercomNames;
         } else {
+            (_target getVariable [QGVAR(unitInfantryPhone), [acre_player, INTERCOM_DISCONNECTED]]) params ["_unitInfantryPhone", ""];
             if (_vehicleInfantryPhone == _target) then {
                 // Generate the action to return the infantry telephone
-                if (_infantryPhoneNetwork == CREW_INTERCOM) then {
-                    private _action = [
-                        "acre_switch_infantryTelephonePassenger",
-                        format [localize LSTRING(switchInfantryPhone), "(" + localize LSTRING(passenger) + ")"],
-                        "",
-                        {
-                            //USES_VARIABLES ["_target", "_player"];
-                            [_target, _player, 3, PASSENGER_INTERCOM] call FUNC(updateInfantryPhoneStatus)
-                        },
-                        {!((_target getVariable [QGVAR(passengerIntercomPositions), []]) isEqualTo [])},  // This should only be available if the vehicle has passenger intercom
-                        {},
-                        {}
-                    ] call ace_interact_menu_fnc_createAction;
-                    _actions pushBack [_action, [], _target];
-                    _action = [
-                        "acre_return_infantryTelephoneCrew",
-                        localize LSTRING(returnInfantryPhone),
-                        "",
-                        {
-                            //USES_VARIABLES ["_target", "_player"];
-                            [_target, _player, 0, CREW_INTERCOM] call FUNC(updateInfantryPhoneStatus)
-                        },
-                        {true}, {}, {}] call ace_interact_menu_fnc_createAction;
-                    _actions pushBack [_action, [], _target];
-                };
+                private _action = [
+                    format ["acre_return_infantryTelephone"],
+                    format [localize LSTRING(returnInfantryPhone)],
+                    "",
+                    {
+                        params ["_target", "_player", "_params"];
+                        _params params ["_intercomNetwork"];
+                        //USES_VARIABLES ["_target", "_player"];
```

Comment:
You have them in `params`, so no need for that. Or remove first `params` and use:
```sqf
(_this select 2) params ["_intercomNetwork"];
```